apps/x509.c

```diff
@@ -1097,3 +1105,93 @@ static int purpose_print(BIO *bio, X509 *cert, X509_PURPOSE *pt)
     }
     return 1;
 }
+
+static int parse_ext_names(char *names, const char **result)
+{
+    char *p, *q;
+    int cnt = 0, idx = 0, len = 0;
+
+    p = q = names;
+    len = strlen(names) + 1;
+
+    while (q - names <= len) {
```

Comment:
The `p != q` check isn't required in the strchr loop:

```
if (*p != '\0') {
    while ((q = strchr(p, ',')) != NULL) {
            if (result != NULL) {
                result[idx++] = p;
                *q = '\0';
            }
            cnt++;
        p = q + 1;
    }

    if (result != NULL)
        result[idx++] = p;
    cnt++;
}
```

It's a matter of taste which reads more easily.  So no need to change it.